src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -112,29 +166,71 @@ public static object ConvertResult(JsonOperationContext ctx, object result)
 
     public class RemoveCompareExchangeCommand : CompareExchangeCommandBase
     {
+        public bool FromBackup;
+
         public RemoveCompareExchangeCommand() { }
-        public RemoveCompareExchangeCommand(string key, string database, long index, JsonOperationContext contextToReturnResult) : base(key, database, index, contextToReturnResult) { }
+        public RemoveCompareExchangeCommand(string database, string key, long index, JsonOperationContext contextToReturnResult) : base(database, key, index, contextToReturnResult) { }
 
-        public override unsafe (long Index, object Value) Execute(TransactionOperationContext context, Table items, long index)
+        public RemoveCompareExchangeCommand(string database, string key, long index, JsonOperationContext contextToReturnResult, bool fromBackup) : base(database, key,
+            index, contextToReturnResult)
+        {
+            FromBackup = fromBackup;
+        }
+
+        public override unsafe (long Index, object Value) Execute(TransactionOperationContext context, Table items, long index, Table tombstoneItems, bool fromBackup = false)
         {
             using (Slice.From(context.Allocator, ActualKey, out Slice keySlice))
             {
                 if (items.ReadByKey(keySlice, out var reader))
                 {
-                    var itemIndex = *(long*)reader.Read((int)ClusterStateMachine.UniqueItems.Index, out var _);
-                    var storeValue = reader.Read((int)ClusterStateMachine.UniqueItems.Value, out var size);
+                    if (fromBackup)
+                    {
+                        items.Delete(reader.Id);
+                        return (index, null);
+                    }
+                    var itemIndex = *(long*)reader.Read((int)ClusterStateMachine.CompareExchangeTable.Index, out var _);
+                    var storeValue = reader.Read((int)ClusterStateMachine.CompareExchangeTable.Value, out var size);
                     var result = new BlittableJsonReaderObject(storeValue, size, context);
+
                     if (Index == itemIndex)
                     {
                         result = result.Clone(context);
                         items.Delete(reader.Id);
+                        WriteCompareExchangeTombstone(context, tombstoneItems, index);
```

Comment:
It is possible to use `context.Transaction.InnerTransaction.OpenTable(CompareExchangeTombstoneSchema, CompareExchangeTombstone)` instead of passing the `tombstoneItems` ?

Reply:
done